Modules/fcntlmodule.c

```diff
@@ -246,17 +250,46 @@ fcntl_ioctl_impl(PyObject *module, int fd, unsigned int code,
         }
 
         PyErr_Clear();
-        if (!PyArg_Parse(ob_arg,
-             "i;ioctl requires a file or file descriptor,"
-             " an integer and optionally an integer or buffer argument",
-             &arg)) {
-          return NULL;
+        if (PyLong_Check(ob_arg)) {
```

Comment:
Actually the `i` code is even more permissive (for example it accepts `Decimal` and `Fraction`). I have restored its logic for compatibility, but later I'm going to restrict the argument to index-like types in master.